### PR TITLE
fix: some type hint

### PR DESCRIPTION
**Pull request by vvanglro**
better type hints.

### CODE REVIEW

File: asyncio_mqtt/client.py [hunk 1]
```diff
@@ -104,16 +105,17 @@ def __init__(
...
 # `nullcontext` when we support Python 3.10 (`nullcontext` becomes async-aware in
 # 3.10). See: https://docs.python.org/3/library/contextlib.html#contextlib.nullcontext
 def _outgoing_call(
-    method: Callable[Concatenate[Client, P], Coroutine[Any, Any, T]]
-) -> Callable[Concatenate[Client, P], Coroutine[Any, Any, T]]:
+    method: Callable[Concatenate[S, P], Coroutine[Any, Any, T]]
+) -> Callable[Concatenate[S, P], Coroutine[Any, Any, T]]:
     @functools.wraps(method)
-    async def decorated(self: Client, *args: P.args, **kwargs: P.kwargs) -> T:
+    async def decorated(self: S, /, *args: P.args, **kwargs: P.kwargs) -> T:
```
Comment (frederikaalund) [inline asyncio_mqtt/client.py:118]: Remove the `/`. PEP 570 (positional-only arguments) is pythton 3.8 and forward. We have to support python 3.7 for now.

File: asyncio_mqtt/client.py [hunk 2]
```diff
@@ -104,16 +105,17 @@ def __init__(
 )
 
 P = ParamSpec("P")
+S = TypeVar("S", bound="Client")
```
Comment (frederikaalund) [inline asyncio_mqtt/client.py:108]: Let's call it `ClientT` to make the use case clearer.

**Review by frederikaalund — CHANGES_REQUESTED** [review]
Thank you for this PR. Let me have a look. :)

I just have some minor comments. With those changes in place, this PR is ready to merge. Good job! :+1: 

I guess the use case here is to support derived classes of `Client`, right?

Best regards,
~Frederik

File: asyncio_mqtt/client.py (hunk 1 above)
Comment (vvanglro) [reply asyncio_mqtt/client.py]: Done.

File: asyncio_mqtt/client.py (hunk 2 above)
Comment (vvanglro) [reply asyncio_mqtt/client.py]: Done.

**Comment by vvanglro** [comment]
Yes, derived classes of Client are supported.

File: asyncio_mqtt/client.py (hunk 2 above)
Comment (vvanglro) [reply asyncio_mqtt/client.py]: Hold on, this push failed, I'm on my way, back to push again

**Comment by frederikaalund** [comment]
LGTM! Cheers. :) Thank you for your contribution to asyncio-mqtt. :+1: 

Regarding the pycharm issue: I'm open to workarounds. Your `@overload`-based workaround is a bit too much for me (defeats the DRY principle). Any other suggestions? :)

**Comment by JonathanPlasse** [comment]
Great PR!
Thanks

**Pull request merged**